torch/csrc/jit/serialization/export_universal.cpp

```diff
@@ -0,0 +1,147 @@
+#include <torch/csrc/jit/serialization/export_universal.h>
+
+#include <torch/csrc/jit/serialization/import_export_helpers.h>
+#include <torch/csrc/jit/serialization/pickler.h>
+#include <torch/csrc/jit/serialization/source_range_serialization.h>
+
+namespace torch {
+namespace jit {
+
+void ScriptModuleSerializerUniversal::serialize(
+    Module& module,
+    const std::string& ts_id) {
+  // Serialize the model object
+  writeArchive("data", module._ivalue(), ts_id);
+  // Then we serialize all code info.
+  convertTypes(module.type());
+  // The tensor constants from the code are written to a separate archive
+  // so loading the code does not depend on loading the data
+  std::vector<IValue> ivalue_constants(
+      constant_table_.begin(), constant_table_.end());
+  writeArchive(
+      "constants", c10::ivalue::Tuple::create(ivalue_constants), ts_id);
+}
+
+void ScriptModuleSerializerUniversal::writeArchive(
+    const std::string& archive_name,
+    const IValue& value,
+    const std::string& ts_id) {
+  std::vector<char> data;
+  // Vector to capture the run-time class types during pickling the IValues
+  std::vector<c10::ClassTypePtr> memoizedClassTypes;
+  // from the exporter
+  std::vector<std::string> tensor_names;
+  Pickler data_pickle(
+      [&](const char* buf, size_t size) {
+        data.insert(data.end(), buf, buf + size);
+      },
+      nullptr,
+      [&](const c10::ClassTypePtr& t) {
+        return type_name_uniquer_.getUniqueName(t);
+      },
+      &memoizedClassTypes,
+      [&]() {
+        // returns a string to use in picker.cpp as storage obj key
+        tensor_names.push_back(getNextStorageID() + ".storage");
```

Comment:
This compiles here but causes compiling issues in this PR: https://github.com/pytorch/pytorch/pull/54903/files/37398f9ed05d74aa79fa5a9807e3b5d1f6ade196#r607995065
I'm not sure what is causing the difference